lib/chat_app/contexts/messages.ex

```diff
@@ -0,0 +1,70 @@
+defmodule ChatApp.Contexts.Messages do
+  @adapter ChatApp.DatabaseAdapter
+  alias ChatApp.Structs.Message
+
+  def to_record(%ChatApp.Structs.Message{
+        id: id,
+        sender: sender,
+        content: content,
+        room_id: room_id,
+        timestamp: timestamp,
+        is_deleted: is_deleted,
+        is_edited: is_edited
+      }) do
+    {id, sender, content, room_id, timestamp, is_deleted, is_edited}
+  end
+
+  def from_record([{id, sender, content, room_id, timestamp, is_deleted, is_edited}]) do
```

Comment:
You are pattern matching against a list, which is not really intuitive in regard to the function name.
I suggest pattern matching against a tuple, and checking the result of the ETS call prior to the call:

```elixir
case ets_result do
    [item] -> from_record(item) # item is a tuple containing the fields mentioned
    [] -> nil
    .... # all the other possible cases (if any)
end
```

Reply:
I did that pattern matching such that I can use both :ets.tab2list and :ets.lookup without any more code, as :ets.lookup also returns a list with all the elements that have the same key. Since this case will never happen for us, can I use instead of checking the ETS call, to modify

`def get_room(id), do: @adapter.get(Room, id) |> from_record()`

Into

`def get_room(id), do: @adapter.get(Room, id) |> Enum.map(fn row -> from_record(row) end)`

And same for messages.

Or do we take the general approach in case our data in a table will have multiple types?